### PR TITLE
feat(configstack): start and finish module logs on run-all execution

### LINKED ISSUE #2216: Feature request: Path output from run-all
In this [issue](https://github.com/gruntwork-io/terragrunt/issues/2170) and the follow up pull request the paths for different states are called output for errors. It would be nice if they were called out in normal operation. This is achievable with `--terragrunt-log-level debug`, but I wouldn't want to turn it on all the time, and I don't want to have to remember to add it just for run-all. A cleaner version of output like this would be awesome. Then a user could tell which state is which in the output at a glance.

``--terragrunt-log-level debug` output:
```
DEBU[0023] Module /home/user/git/terraform/aws/development/us-east-2 has finished successfully!  prefix=[/home/user/git/terraform/aws/development/us-east-2] 
```

### PR DESCRIPTION
**Pull request by juanesech**
## Description
Log when a module start/finish on multi-modules execution for an easier visualization of the terraform commands and their results

Fixes #2216.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

**Review by Etiene — APPROVED** [review]
Thanks for the PR! Code looks good to me, pending one more approval from an appointed maintainer.

**Comment by yhakbar** [comment]
@juanesech 

Sorry we didn't get to this PR before it went stale. We have significant plans in the works for improving logging in Terragrunt, so know that there will be improvements in the near future for addressing this need.

Closing PR as stale.

**Pull request closed without merging**
